Add service URL documentation (#7914)
qgis/QGIS-Documentation: docs/server_manual/config.rst

     - filesystem
     - All

   * - .. _qgis_server_service_url:

       QGIS_SERVER_SERVICE_URL
     - This is an option to set the service URL if it is not present in the project.

       The service URL is defined from (in order of precedence):

       * Value defined in the project per service
       * The ``QGIS_SERVER_<service>_SERVICE_URL`` environment variable
       * The ``QGIS_SERVER_SERVICE_URL`` environment variable
       * The ``X-Qgis-<service>-Service-Url`` header
       * The ``X-Qgis-Service-Url`` header
       * Build from the ``Forwarded`` header
       * Build from the ``X-Forwarded-Host`` and ``X-Forwarded-Proto`` headers
       * Build from the ``Host`` header and the server protocol
       * Build from the server name and the server protocol.

       The last three also use the ``REQUEST`` path and the ``MAP`` provided

>        in the query string.

Reviewer: @sbrunner @pblottiere just wonder if this sentence is correct. Does REQUEST have path or is it not MAP that is a path?

Author: This sentence should effectively be rewritten because she is hard to understand...
In something like:
The built URL (the fore last cases) will keep the provided path and the map parameter provided by the query string.

Reviewer: ```suggestion
       The built URL in the last four cases will keep the provided path and
       the ``MAP`` parameter provided by the query string.
```
"Path provided" refers to service_URL value?

Reviewer 2: What do you think about something like:

*In the last four cases, the resulting Service URL is based on the ``MAP`` parameter provided in the query string and on the incoming path request.*

Reviewer 2: @DelazJ I think we can apply my suggestion ^

Reviewer 2: (if you agree with the wording of course)

     - ''
     - All

   * - SERVICE_URL

>      - This is an option To set the service URL it it's not present in the project.

Reviewer 2: ```suggestion
     - This is an option to set the service URL if it's not present in the project.
```

>        The service URL resolution watch on the following and thaks the first:

Reviewer 2: Typo (`thaks`)

         - Value defined in the project per service.
         - The ``<service>_SERVICE_URL`` environment variable.
         - The ``SERVICE_URL`` environment variable.

>          - The ``X-<service>-Servcie-Url`` header.

Reviewer 2: ```suggestion
         - The ``X-<service>-Service-Url`` header.
```

         - The ``X-Service-Url`` header.
         - Build form the ``Forwarded`` header.
         - Build form the ``X-Forwarded-Host`` and ``X-Forwarded-Proto`` headers.
         - Build form the ``Host`` header and the server protocol.
         - Build form the server name and the server protocol.

>        The tree last also use the request path and the ``MAP`` provided in the query string.

Reviewer 2: ```suggestion
       The three last also use the request path and the ``MAP`` provided in the query string.
```

     - ''
     - All

   * - WMS_SERVICE_URL

>      - This is an option To set the service URL it it's not present in the project.

Reviewer 2: ```suggestion
     - This is an option to set the service URL if it's not present in the project.
```

>        See ``SERVICE_URL`` for more informations

Reviewer 2: ```suggestion
       See ``SERVICE_URL`` for more information
```

     - ''
     - WMS

   * - WFS_SERVICE_URL

>      - This is an option To set the service URL it it's not present in the project.

Reviewer 2: ```suggestion
     - This is an option to set the service URL if it's not present in the project.
```

>        See ``SERVICE_URL`` for more informations

Reviewer 2: ```suggestion
       See ``SERVICE_URL`` for more information
```

     - ''
     - WFS

   * - WMTS_SERVICE_URL

>      - This is an option To set the service URL it it's not present in the project.

Reviewer 2: ```suggestion
     - This is an option to set the service URL if it's not present in the project.
```

>        See ``SERVICE_URL`` for more informations

Reviewer 2: ```suggestion
       See ``SERVICE_URL`` for more information
```

     - ''
     - WMTS

   * - WCS_SERVICE_URL

>      - This is an option To set the service URL it it's not present in the project.

Reviewer 2: ```suggestion
     - This is an option to set the service URL if it's not present in the project.
```

>        See ``SERVICE_URL`` for more informations

Reviewer 2: ```suggestion
       See ``SERVICE_URL`` for more information
```
